### ISSUE TITLE
"Inline Variable" assist should trigger on variable uses

### ISSUE THREAD

**Issue by jonas-schievink** [rust-analyzer/rust-analyzer#8530]
```rust
fn f() {
    let variable$0 = "";
    variable$1;
}
```

Currently, this only shows the "inline variable" assist at `$0`. Ideally it would also show up at `$1`, but only replace that specific use, not all of them (but still remove the `let` statement if that is the only use).
